test_harness.jl

```diff
@@ -4,7 +4,8 @@ kwargs = Kwargs.kwargs(; coverage=ENV["COVERAGE"],
                          force_latest_compatible_version=ENV["FORCE_LATEST_COMPATIBLE_VERSION"],
                          allow_reresolve=ENV["ALLOW_RERESOLVE"],
                          julia_args=[string("--check-bounds=", ENV["CHECK_BOUNDS"]),
-                                     string("--compiled-modules=", ENV["COMPILED_MODULES"])],
+                                     string("--compiled-modules=", ENV["COMPILED_MODULES"]),
+                                     string("--depwarn=", ENV["DEPWARN"]),],
```

Comment:
Best to leave a breadcrumb for why we are taking this approach:
```suggestion
                                     # Needs to be done via `julia_args` to ensure `depwarn: no` is respected:
                                     # https://github.com/JuliaLang/Pkg.jl/pull/1763#discussion_r406819660
                                     string("--depwarn=", ENV["DEPWARN"]),],
```